### PR TITLE
[libdwarf] Update to v0.9.1

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

This PR depends on #36524

### CODE REVIEW

**Comment by dg0yt** [comment]
If there is v0.9.1, v0.9.0 could be skipped.

File: versions/c-/cpptrace.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "a9dc1989a430b531df437790e61d09b411342ec3",
+      "git-tree": "0c8ce6d1bfc9c8c519e96079e11fd380752266dd",
```
Comment (jimwang118) [inline versions/c-/cpptrace.json:4]: You should add "port-version: 1" to cpptrace/vcpkg.json.

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: versions/c-/cpptrace.json (hunk 1 above)
Comment (jeremy-rifkin) [reply versions/c-/cpptrace.json:4]: Initially I did change the port-version but I was getting CI errors along the lines of `Error: PRs must add only one version, and must not modify any published versions.` and `Error: After committing all other changes, the version database must be updated.`. Are those to be expected?

Comment (jimwang118) [reply versions/c-/cpptrace.json:4]: Yes, you should add "port-version: 1" in `cpptrace/vcpkg.json`. Then run the command: `./vcpkg x-add-version cpptrace`.

Comment (jeremy-rifkin) [reply versions/c-/cpptrace.json]: Thanks!

**Comment by jeremy-rifkin** [comment]
Alright, I think this should be ready for review

File: versions/c-/cpptrace.json [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-      "git-tree": "a9dc1989a430b531df437790e61d09b411342ec3",
+      "git-tree": "bccfdc44eade3f68da43360a94d04e31a0fd94d1",
       "version": "0.3.1",
       "port-version": 0
     },
+    {
+      "git-tree": "0c8ce6d1bfc9c8c519e96079e11fd380752266dd",
+      "version": "0.3.1",
+      "port-version": 1
+    },
```
Comment (jimwang118) [inline versions/c-/cpptrace.json:12]:
```suggestion
```
You don’t need to manually modify this file. You need to modify the `vcpkg.json` file under `cpptrace`, add port-version:1 to the `vcpkg.json` file, then run `./vcpkg x-add-verison cpptrace`, and submit the results after the command is completed.
You can refer to how [port-version](https://github.com/microsoft/vcpkg-docs/blob/c40eb4fbc629d2f357dfc6528d03703471968454/vcpkg/users/versioning.md#-port-version) is modified.

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: versions/c-/cpptrace.json (hunk 2 above)
Comment (jeremy-rifkin) [reply versions/c-/cpptrace.json]: Oh sorry, thanks!

**Comment by jeremy-rifkin** [comment]
CI doesn't seem happy with the port version stuff for cpptrace, not sure what's wrong but I think I did the ./vcpkg x-add-verison cpptrace correctly this time

**Comment by jimwang118** [comment]
You need to run the `./vcpkg x-add-verison cpptrace --overwrite-version` command and then submit.

**Comment by jeremy-rifkin** [comment]
Thanks, looks like that worked

**Review by jimwang118 — APPROVED** [review]

**Comment by jimwang118** [comment]
Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

**Review by data-queue — APPROVED** [review]

**Comment by VelocityRa** [comment]
This PR replaced the previous version (0.8.0) with the new one in `versions/l-/libdwarf.json`. I don't think that's supposed to happen? All versions should be present. I'm getting errors for version 0.8.0 now.

**Comment by jimwang118** [comment]
> This PR replaced the previous version (0.8.0) with the new one in `versions/l-/libdwarf.json`. I don't think that's supposed to happen? All versions should be present. I'm getting errors for version 0.8.0 now.

I restored the version in PR [37367](https://github.com/microsoft/vcpkg/pull/37367), you can test it locally.

**Pull request merged**